ext/ClimaAnalysisExt.jl

```diff
@@ -546,23 +547,55 @@ function ObservationRecipe.reconstruct_g_mean_final(
 
     # Check if length of g ensemble is the same as the length of the data in the metadatas
     total_metadata_length =
-        sum(ClimaAnalysis.Var._data_size(metadata) for metadata in metadatas)
+        sum(metadata_length(metadata) for metadata in metadatas)
     length(g_mean) != total_metadata_length && error(
         "Length of g_mean_final is not the same as the length of all the metadata",
     )
 
     # Reconstruct each OutputVar from the metadata
+    minibatch_indices = _get_minibatch_indices_for_nth_iteration(
+        obs_series,
+        EKP.get_N_iterations(ekp),
+    )
+    vars = map(metadatas, minibatch_indices) do metadata, range
+        ClimaAnalysis.unflatten(metadata, g_mean[range])
+    end
+    return vars
+end
+
+"""
+    _get_minibatch_indices_for_nth_iteration(ekp::EKP.EnsembleKalmanProcess, N)
+
+Get the indices that correspond to each metadata for the minibatch of the `N`th
+iteration.
+
+Note that the `indices` field in the `EKP.observation` cannot be used as the
+multiple `OutputVar`s are flattened and concatenated together as a single
+vector.
+"""
+function _get_minibatch_indices_for_nth_iteration(obs_series, N)
```

Comment:
It would be nice to expose this info to the user so they can see the mapping of indices to metadata/short names, it could be a helper function called something like `ranges_by_short_name`

Reply:
Yes, it should take a short name and return the indices, it might be useful but I don't think it's necessary in this PR.